pkg/executor/aggfuncs/spill_serialize_helper.go

```diff
@@ -0,0 +1,257 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aggfuncs
+
+import (
+	"unsafe"
+
+	"github.com/pingcap/tidb/pkg/types"
+	"github.com/pingcap/tidb/pkg/util/spill"
+)
+
+// SpillSerializeHelper can only be used for one aggregator function.
+// It may cause error if agg func1 and agg func2 use the same SpillSerializeHelper.
+type SpillSerializeHelper struct {
+	buf []byte
+}
+
+// NewSpillSerializeHelper creates a new SpillSerializeHelper
+func NewSpillSerializeHelper() *SpillSerializeHelper {
+	return &SpillSerializeHelper{
+		buf: make([]byte, 1024),
+	}
+}
+
+func (s *SpillSerializeHelper) serializePartialResult4Count(value partialResult4Count) []byte {
+	return spill.SerializeInt64(value, s.buf[0:int64Len])
+}
+
+func (s *SpillSerializeHelper) serializePartialResult4MaxMinInt(value partialResult4MaxMinInt) []byte {
+	spill.SerializeInt64(value.val, s.buf[0:int64Len])
+	end := int64Len + boolLen
+	spill.SerializeBool(value.isNull, s.buf[int64Len:end])
+	return s.buf[0:end]
+}
```

Comment:
I doubt whether we need to take the xxLen into consideration in `serializeXXX`, it'll make the serialize funcs too complicated. We only need to consider the xxLen when deserializing.
e.g.
```suggestion
func (s *SpillSerializeHelper) serializePartialResult4MaxMinInt(value partialResult4MaxMinInt) []byte {
	s.buf = spill.SerializeInt64(value.val, s.buf)
	s.buf = spill.SerializeBool(value.isNull, s.buf)
	return s.buf
}

func SerializeInt64(value int64, buf []byte) []byte {
	var data [8]byte
	binary.BigEndian.PutUint64(data[:], uint64(value))
	buf = append(buf, data[:]...)
	return buf
}

func SerializeFloat64(value float64, buf []byte) []byte{
        ...
        binary.BigEndian.PutUint64(data[:], math.Float64bits(value))

        ...
}

```

Reply:
> I doubt whether we need to take the xxLen into consideration in `serializeXXX`, it'll make the serialize funcs too complicated. We only need to consider the xxLen when deserializing. e.g.

I think xxxLen is necessary as the format of serialize and deserialize should be the same which is guaranteed by xxxLen.

Reply:
got

I think it's better to simplify and unify the interface definition and usage of the SerializeX funcs. It'll be more readable.
e.g.
``` go
func SerializeX(value Any, buf []byte) []byte{
        buf = serializeY()
        buf = serializeZ()
        return buf
}
```

Reply:
> got
> 
> I think it's better to simplify and unify the interface definition and usage of the SerializeX funcs. It'll be more readable. e.g.
> 
> ```go
> func SerializeX(value Any, buf []byte) []byte{
>         buf = serializeY()
>         buf = serializeZ()
>         return buf
> }
> ```

done